base/boot.jl

```diff
@@ -195,39 +195,38 @@ end
 Expr(@nospecialize args...) = _expr(args...)
 
 abstract type Exception end
-mutable struct ErrorException <: Exception
+struct ErrorException <: Exception
     msg::AbstractString
-    ErrorException(msg::AbstractString) = new(msg)
 end
 
 macro _noinline_meta()
     Expr(:meta, :noinline)
 end
 
-struct BoundsError        <: Exception
+struct BoundsError <: Exception
     a::Any
     i::Any
     BoundsError() = new()
     BoundsError(@nospecialize(a)) = (@_noinline_meta; new(a))
     BoundsError(@nospecialize(a), i) = (@_noinline_meta; new(a,i))
 end
-struct DivideError        <: Exception end
-struct OutOfMemoryError   <: Exception end
-struct ReadOnlyMemoryError<: Exception end
-struct SegmentationFault  <: Exception end
-struct StackOverflowError <: Exception end
-struct UndefRefError      <: Exception end
-struct UndefVarError      <: Exception
+struct DivideError         <: Exception end
+struct OutOfMemoryError    <: Exception end
+struct ReadOnlyMemoryError <: Exception end
+struct SegmentationFault   <: Exception end
+struct StackOverflowError  <: Exception end
+struct UndefRefError       <: Exception end
+struct UndefVarError <: Exception
     var::Symbol
 end
 struct InterruptException <: Exception end
 struct DomainError <: Exception
     val
-    msg
-    DomainError(@nospecialize(val)) = (@_noinline_meta; new(val))
```

Comment:
cc @timholy  is this supposed to leave the `msg` field to be `#undef` ? Seems better to change it to an empty string as I have done in this PR.